### PR TITLE
Set the type of batch jobs to FNAPI_BATCH when beam_fn_api experiment…

### PR DESCRIPTION
**Pull request by tvalentyn**
… is specified.

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by tvalentyn** [comment]
R: @vikkyrk 
CC: @aaltay

**Comment by tvalentyn** [comment]
CC: @robertwb

**Comment by vikkyrk** [comment]
LGTM

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient.py [hunk 1]
```diff
@@ -146,7 +146,14 @@ def __init__(self, packages, options, environment_version):
     if self.standard_options.streaming:
       job_type = 'FNAPI_STREAMING'
     else:
-      job_type = 'PYTHON_BATCH'
+      use_fnapi_for_batch = (
+          self.debug_options.experiments
+          and 'beam_fn_api' in self.debug_options.experiments
+      )
+      if use_fnapi_for_batch:
+        job_type = 'FNAPI_BATCH'
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/dataflow/internal/apiclient.py]: Does it require a change in the `dependency.py`. Or is `BEAM_FNAPI_CONTAINER` is the correct container for both batch and streaming fn api jobs?

Comment (tvalentyn) [inline sdks/python/apache_beam/runners/dataflow/internal/apiclient.py]: Batch and streaming use the same container, the logic is already in dependency.py: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/runners/dataflow/internal/dependency.py#L501, https://github.com/apache/beam/blob/master/sdks/python/apache_beam/runners/dataflow/internal/dependency.py#L533.

**Comment by tvalentyn** [comment]
Closed in favor of https://github.com/apache/beam/pull/3522

**Pull request closed without merging**